zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompileTask.kt

```diff
@@ -88,19 +89,33 @@ abstract class ZiplineCompileTask @Inject constructor(
       else -> null
     }
 
-    ZiplineCompiler.compile(
-      inputDir = inputDirFile,
-      outputDir = outputDirFile,
-      mainFunction = mainFunction,
-      mainModuleId = mainModuleId,
-      manifestSigner = manifestSigner
-    )
+    if (inputChanges.isIncremental) {
+      val filterByChangeType: ((ChangeType) -> Boolean) -> List<File>  = { filter ->
+        inputChanges.getFileChanges(inputDir)
```

Comment:
```suggestion
      fun filterByChangeType(filter: (ChangeType) -> Boolean): List<File> {
        return inputChanges.getFileChanges(inputDir)
```
Just _ever_ so slightly more efficient but also easier to visually parse, in my opinion.